### ISSUE TITLE
Add-KeeperSecretManagerClient can't return base64 configuration string

### ISSUE THREAD

**Issue by sk-keeper** [Keeper-Security/secrets-manager#539]
https://github.com/Keeper-Security/keeper-sdk-dotnet/issues/103

PowerCommander doesn't seem to have the option to return a base64 configuration string when creating a Keeper Secret Manager Client. Only the one time access token is provided as a return value. Example:
Add-KeeperSecretManagerClient -App “someApp”

Whereas, the regular Commader CLI can do the following:
secrets-manager client add --app someApp --config-init b64

Could you please add the ability to create base64 configs in the PowerCommander Add-KeeperSecretManagerClient function?

Alternatively, is there a current method in PowerCommander to convert a one time access token into a base64 config?

Extra Context
I am evaluating the Secret Manager product as a possible purchase add-on for my organization (we are a current Keeper customer).

Here's what I am trying to accomplish:

During an automated Infrastructure Deployment...

Create Shared Folders, Applications and Device Clients with [PowerCommander](https://docs.keeper.io/secrets-manager/commander-cli/commander-installation-setup/installation-on-powershell)
Invoke [Keeper Terraform](https://registry.terraform.io/providers/Keeper-Security/secretsmanager/latest) to create secrets for the Shared Folders (your terraform module requires a configuration, which is why I need the b64 value)
Use SecretsManager [PowerShell plugin](https://docs.keeper.io/secrets-manager/secrets-manager/integrations/powershell-plugin) to pull secrets as needed (also using b64 config value).

**Comment by rdavisunr** [comment]
Hello, I originally opened this issue in the other repo.

I wanted to provide a code snippet that shows how I got around the current limitation:

```powershell
Write-Host "Creating Shared Folder $envTenant"

# The shared folder (per environment-tenant-name)
# NOTE: This will throw an exception if the folder already exists
$folderInfo = Add-KeeperFolder -Name $envTenant -Shared

# The seed secret (per tenant-environment)
# the folder must contain a secret before Terraform can add Records to the folder
Add-KeeperRecord -Title "Seed" -Folder $folderInfo.Name -GeneratePassword | Out-Null

Write-Host "Adding Application and Client"

# The deployment “application” for the Tenant
Add-KeeperSecretManagerApp -AppName $applicationName | Out-Null

# The deployment client for the Tenant
# NOTE: Add-KeeperSecretManagerClient won't return a b64 
# configuration, thus requiring a second call to the KSM CLI below
# Related GitHub issue: https://github.com/Keeper-Security/secrets-manager/issues/539
$oneTimeAccessCode = Add-KeeperSecretManagerClient -App $applicationName -Name $applicationDeviceName -UnlockIp 

Write-Host "Granting Application Access to Shared Folder"

# Sharing the shared folder to the application
# NOTE: -Secret can be the Shared Folder Name or Shared Folder UID
Grant-KeeperSecretManagerFolderAccess -App $applicationName -Secret $folderInfo.FolderUid -CanEdit | Out-Null

Write-Host "Calling Keeper Secrets Manager CLI for one-time token exchange"

$ksmCliPath = Join-path -Path ${env:ProgramFiles(x86)} -ChildPath "Keeper Secrets Manager CLI\keeper-ksm.exe"

$b64Code = & "$ksmCliPath" init default $oneTimeAccessCode

$value = "" | Select-Object -Property SharedFolderUid, SharedFolderName, Base64Configuration
$value.SharedFolderUid = $folderInfo.FolderUid
$value.SharedFolderName = $folderInfo.Name
$value.Base64Configuration = $b64Code

return $value
```

This workaround means the KSM CLI has to be present on our automation agent(s).

Ideally, it would be nicer if PowerCommander's `Add-KeeperSecretManagerClient` cmdlet could just return the b64 encoded config in the same manner that the regular Commander CLI can; for example: `secrets-manager client add --app someApp --config-init b64`

**Comment by rdavisunr** [comment]
@idimov-keeper It seems like there is some confusion here...

I am currently using **PowerCommander** to create BOTH Applications and Clients. My ask is that PowerCommander's `Add-KeeperSecretManagerClient` function be updated to allow the base64 config to be returned (it currently only returns the one-time-token). I am only using KSM in the code sample above because PowerCommander lacks that functionality (but as you point out, that ability is available in Keeper Commander). I am asking that the ability also be available in PowerCommander.

I think the issue here is that @sk-keeper moved this issue from the [keeper-sdk-dotnet](https://github.com/Keeper-Security/keeper-sdk-dotnet) repo (where I originally logged it) to this repo.

But, seems like the original repo (where PowerCommander lives) was the correct spot for this issue. Does that sound correct?

**Comment by rdavisunr** [comment]
Agreed - my original hope was not to use KSM at all for this use case. I am only using it because the token exchange isn't available in PowerCommander.

I just added this comment on the original issue in the keeper-sdk-dotnet repo:
https://github.com/Keeper-Security/keeper-sdk-dotnet/issues/103#issuecomment-1802882350

I hope that explains the request with better clarity.
